### PR TITLE
WIP: CSV optional column names

### PR DESCRIPTION
**Pull request by JasiekChomiak**
The implementation seems to work, but I haven't tested it yet that much (and haven't written any tests) so it's WIP

This solves #133

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ JSON file in one of the following forms:
 CSV file seperated using commas. The first row should contain column names.
 ##### options:
 - path - path to file containing the data, required
-
+- columns - optional, defaults to true. If false then column names are col1, col2...
```
Comment (cube2222) [inline README.md]: Please change this to something like:
containsHeaderRow, or headerRow

File: storage/csv/datasource.go [hunk 2]
```diff
@@ -58,9 +62,35 @@ func (ds *DataSource) Get(variables octosql.Variables) (execution.RecordStream,
...
+
+		columnCount := len(firstRow)
+
+		columns = make([]string, columnCount)
+
+		for i := 0; i < columnCount; i++ {
+			columns[i] = fmt.Sprintf("col%d", i+1)
+		}
+
+		_, err = file.Seek(0, 0) /* we must return to the beginning of file */
```
Comment (cube2222) [inline storage/csv/datasource.go]: Why? The first row was the header row, we don't want it anymore

Comment (JasiekChomiak) [reply storage/csv/datasource.go]: It's actually the opposite. There is no header row, so we read the first one, to determine how many columns there are (to create column names col1, col2...). Alternatively the first row could be passed to the stream as a []string, same as in Excel. Don't know which method is better tough

Comment (cube2222) [reply storage/csv/datasource.go]: As discussed in private, I think it's better to read and save the column list on first record read.

File: storage/csv/datasource.go [hunk 3]
```diff
@@ -60,56 +64,107 @@ func (ds *DataSource) Get(variables octosql.Variables) (execution.RecordStream,
...
+	alias           string
+	aliasedFields   []octosql.VariableName
+	first           bool
+	hasColumnHeader bool
+}
+
+func (rs *RecordStream) Close() error {
+	err := rs.file.Close()
+	if err != nil {
+		return errors.Wrap(err, "Couldn't close underlying file")
```
Comment (cube2222) [inline storage/csv/datasource.go]: errors should start lowercase

**Pull request merged**